### PR TITLE
LPS-76663 Message boards orders inccorectly

### PR DESCRIPTION
**Pull request by hudakl**
Hi Serigo,

Please find the solution for LPS-76663 in this pull request.
Could you please review the changes?

Thanks,
Laci

### CODE REVIEW

**Comment by hudakl** [comment]
Hi Sergio,

Do you have any updates on this pull?

Thanks,
Laci

File: portal-impl/src/com/liferay/portlet/messageboards/service/impl/MBMessageLocalServiceImpl.java [hunk 1]
```diff
@@ -1556,9 +1556,9 @@ public MBMessage updateMessage(
 
 			thread.setPriority(priority);
 
-			mbThreadPersistence.update(thread);
-
 			updatePriorities(thread.getThreadId(), priority);
+
```
Comment (sergiogonzalez) [inline portal-impl/src/com/liferay/portlet/messageboards/service/impl/MBMessageLocalServiceImpl.java]: Hey @hudakl we should undo this change and keep the original order. First update persistence and then update priorities, since the fix for the bug is not related with reordering this.

**Comment by sergiogonzalez** [comment]
Can we introduce a integration test to ensure the behaviour is working as expected? Ideally we should have a test with workflow and a test without workflow.

**Comment by hudakl** [comment]
Hi Sergio,

It appears that keeping the original order indeed doesn't affect the end result. I can send a new pull where the two commits are merged into one plus another one for the tests. But I'm afraid it could take some time since I'm not really familiar with integration tests.

Thanks,
Laci

**Comment by sergiogonzalez** [comment]
Hey @hudakl 
Feel free to ask @robertoDiaz if you have doubts or questions about how to write those tests.
We should write it here https://github.com/liferay/liferay-portal/blob/master/modules/apps/collaboration/message-boards/message-boards-test/src/testIntegration/java/com/liferay/message/boards/service/test/MBMessageLocalServiceTest.java
Thanks!
